### PR TITLE
Finetuning

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Finetuning, bugfixing and refactoring sprint 18.

### CODE REVIEW

File: CDP4Reporting.Tests/DataCollection/DataCollectorStateDependentRowDoubleParameterTestFixture.cs [hunk 1]
```diff
@@ -115,5 +124,31 @@ public void VerifyThatParametersWithoutStateDependenciesWork()
...
+                .Build();
+
+            var dataSource = new DataCollectorNodesCreator<ErrorRow>();
+            var nestedElementTree = new NestedElementTreeGenerator().Generate(this.dataCollectorParameterTestFixture.option).ToList();
+
+            Assert.Throws<InvalidOperationException>(() => dataSource.CreateNodes(hierarchy, nestedElementTree));
+        }
     }
-}
+}
```
Comment (cozminvelciu) [inline CDP4Reporting.Tests/DataCollection/DataCollectorStateDependentRowDoubleParameterTestFixture.cs:154]: Missing newline at end of file.

File: CDP4Reporting/DataCollection/DataCollectorCategory.cs [hunk 2]
```diff
@@ -88,9 +88,6 @@ internal override void Initialize(DataCollectorNode<T> node, PropertyInfo proper
         /// Populates with data the <see cref="DataTable.Columns"/> associated with this object
         /// in the given <paramref name="row"/>.
         /// </summary>
-        /// <param name="table">
```
Comment (cozminvelciu) [inline CDP4Reporting/DataCollection/DataCollectorCategory.cs:91]: The parameter named "table" is still present in the signature.

File: CDP4Reporting/DataCollection/DataCollectorNode.cs [hunk 3]
```diff
@@ -285,31 +313,20 @@ public DataTable GetTable(bool excludeMissingParameters = false)
...
-            this.AddDataRows(table, excludeMissingParameters);
-
-            return table;
         }
 
         /// <summary>
         /// Gets the row representation of this node.
         /// </summary>
+        /// <param name="forceNew">
+        /// If set to true, a new <see cref="T"/> will be created and the cach field <see cref="rowRepresentation"/> will not be set.
```
Comment (cozminvelciu) [inline CDP4Reporting/DataCollection/DataCollectorNode.cs:322]: Typo: "cach".

File: CDP4Reporting/DataCollection/DataCollectorNode.cs [hunk 4]
```diff
@@ -389,59 +411,120 @@ internal void AddDataRows(DataTable table, bool excludeMissingParameters)
...
+
+                this.FillCategoryDecompositionHierarchyColumns(dataRow);
+                this.FillNormalColumns(rowPresentation, dataRow);
+                this.FillPublicGetterColumns(rowPresentation, dataRow);
             }
+        }
 
-            var removeOriginalColumn = false;
+        /// <summary>
+        /// Fills the <see cref="DataColumn"/>s with data for properties of an instance of <see cref="T"/> whose <see cref="Type"/> implement <see cref="IDataCollectorStateDependentPerRowParameter"/>.
```
Comment (cozminvelciu) [inline CDP4Reporting/DataCollection/DataCollectorNode.cs:432]: This line is a bit too long, I would suggest splitting it before "implement <see ...".

**Review by cozminvelciu — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]

**Review by samatrhea — APPROVED** [review]
no comment... ah, that was already a comment, nevermind

**Pull request merged**